src/Orleans/Serialization/OrleansJsonSerializer.cs

```diff
@@ -1,27 +1,23 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Linq;
 using System.Net;
-using System.Runtime.Serialization.Formatters.Binary;
-using System.Text;
-using System.Threading.Tasks;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
 using Orleans.Runtime;
 
 namespace Orleans.Serialization
 {
-    internal class OrleansJsonSerializer : IExternalSerializer
+    public class OrleansJsonSerializer : IExternalSerializer
     {
-        public static JsonSerializerSettings settings;
+        public static JsonSerializerSettings SerializerSettings { get; set; }
```

Comment:
The point was to allow the user to change the json setting at runtime, for example from bootstrap provider.
I do believe allowing the user to change the json setting at runtime is eventually inevitable, given the unexpected behavior of json serializer (see https://github.com/dotnet/orleans/issues/1187#issuecomment-166769680 which caused me to open this PR).

>Is it desirable for the settings to be flipped back and forth at run time?